### PR TITLE
Add additional 3rd party CAN devices

### PR DESCRIPTION
**Pull request by Spaceman113138**
Adds the grapple lasercan and Redux canandgyro. Also renames the canandcoder to its new name the canandgyro. Should I also remove the canandcolor since there no longer seems to be any documentation on it and the links on the page appear to be broken?

### CODE REVIEW

File: source/docs/software/can-devices/third-party-devices.rst [hunk 1]
```diff
@@ -107,16 +107,32 @@ Playing With Fusion (PWF) offers the Venom integrated motor/controller as well a
 
 ## Redux Robotics
 
-Redux Robotics currently offers the Canandcoder :term:`CAN` + :term:`PWM` magnetic encoder and the Canandcolor :term:`CAN`-enabled color sensor.
+Redux Robotics currently offers the Canandcoder :term:`CAN` + :term:`PWM` magnetic encoder, the Canandcolor :term:`CAN`-enabled color sensor, and the Canandgyro :term:`CAN`-enabled gyro.
```
Comment (shueja) [inline source/docs/software/can-devices/third-party-devices.rst:110]:
```suggestion
Redux Robotics currently offers the HELIUM Canandmag :term:`CAN` + :term:`PWM` magnetic encoder, the Canandcolor :term:`CAN`-enabled color sensor, and the Canandgyro :term:`CAN`-enabled gyro.
```

File: source/docs/software/can-devices/third-party-devices.rst [hunk 2]
```diff
@@ -107,16 +107,32 @@ Playing With Fusion (PWF) offers the Venom integrated motor/controller as well a
...
-    - [Technical Manual](https://docs.reduxrobotics.com/canandcoder/index.html)
+    - API Documentation ([Java](https://apidocs.reduxrobotics.com/current/java/com/reduxrobotics/sensors/canandmag/package-summary), [C++](https://apidocs.reduxrobotics.com/current/cpp/namespaceredux_1_1sensors_1_1canandmag))
+    - [Technical Manual](https://docs.reduxrobotics.com/canandmag)
 
 - **Canandcolor**
 
     - API Documentation ([Java](https://apidocs.reduxrobotics.com/current/java/com/reduxrobotics/sensors/canandcolor/Canandcolor.html), [C++](https://apidocs.reduxrobotics.com/current/cpp/classredux_1_1sensors_1_1canandcolor_1_1Canandcolor.html))
     - [Technical Manual](https://docs.reduxrobotics.com/canandcolor/index.html)
+
+- **BORON CANandgyro**
```
Comment (shueja) [inline source/docs/software/can-devices/third-party-devices.rst:124]:
```suggestion
- **BORON Canandgyro**
```

**Review by shueja — CHANGES_REQUESTED** [review]

**Comment by Spaceman113138** [comment]
The link check likely fails because of how redux configures their webpages to prevent robots from accessing them, hence the 403 no permission instead of a 404 page not found. What would be the best way to proceed?

**Comment by sciencewhiz** [comment]
> The link check likely fails because of how redux configures their webpages to prevent robots from accessing them, hence the 403 no permission instead of a 404 page not found. What would be the best way to proceed?

merge/rebase on main. #2824

File: source/docs/software/can-devices/third-party-devices.rst [hunk 3]
```diff
@@ -19,56 +19,56 @@ CTR Electronics (CTRE) offers several CAN peripherals with external libraries. G
...
-    - [Hardware User's Manual](https://store.ctr-electronics.com/content/user-manual/Falcon%20500%20User%20Guide.pdf)
-    - Software Documentation ([v5](https://v5.docs.ctr-electronics.com/en/stable/ch13_MC.html), [Pro](https://pro.docs.ctr-electronics.com/en/stable/docs/api-reference/examples/quickstart.html))
+    - API Documentation (v6: [Java](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/TalonFX.html), [C++](https://api.ctr-electronics.com/phoenix/release/cpp/classctre_1_1phoenix_1_1motorcontrol_1_1can_1_1_talon_f_x.html) | Pro: [Java](https://api.ctr-electronics.com/phoenixpro/release/java/com/ctre/phoenixpro/hardware/TalonFX.html), [C++](https://api.ctr-electronics.com/phoenixpro/release/cpp/classctre_1_1phoenixpro_1_1hardware_1_1_talon_f_x.html))
```
Comment (sciencewhiz) [inline source/docs/software/can-devices/third-party-devices.rst:24]: the pro API links should be removed. They were superseded by v6.

File: source/docs/software/can-devices/third-party-devices.rst [hunk 4]
```diff
@@ -19,56 +19,56 @@ CTR Electronics (CTRE) offers several CAN peripherals with external libraries. G
...
     - [Software Documentation](https://v5.docs.ctr-electronics.com/en/stable/ch13_MC.html)
 
 ### CTRE Sensors
 
 - **CANcoder**
 
-    - API Documentation (v5: [Java](https://api.ctr-electronics.com/phoenix/release/java/com/ctre/phoenix/sensors/WPI_CANCoder.html), [C++](https://api.ctr-electronics.com/phoenix/release/cpp/classctre_1_1phoenix_1_1sensors_1_1_w_p_i___c_a_n_coder.html) | Pro: [Java](https://api.ctr-electronics.com/phoenixpro/release/java/com/ctre/phoenixpro/hardware/CANcoder.html), [C++](https://api.ctr-electronics.com/phoenixpro/release/cpp/classctre_1_1phoenixpro_1_1hardware_1_1_c_a_ncoder.html))
+    - API Documentation (v6: [Java](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/CANcoder.html), [C++](https://api.ctr-electronics.com/phoenix6/release/cpp/namespacectre_1_1phoenix_1_1platform.html#ad0cac06e0817cbcfe20851a8619a19a8a13b25ca6c3f7037c0f6e13b59cc1e571) | Pro: [Java](https://api.ctr-electronics.com/phoenixpro/release/java/com/ctre/phoenixpro/hardware/CANcoder.html), [C++](https://api.ctr-electronics.com/phoenixpro/release/cpp/classctre_1_1phoenixpro_1_1hardware_1_1_c_a_ncoder.html))
```
Comment (sciencewhiz) [inline source/docs/software/can-devices/third-party-devices.rst:44]: ditto

File: source/docs/software/can-devices/third-party-devices.rst [hunk 5]
```diff
@@ -19,56 +19,56 @@ CTR Electronics (CTRE) offers several CAN peripherals with external libraries. G
...
+    - [Software Documentation](https://v6.docs.ctr-electronics.com/en/stable/docs/hardware-reference/cancoder/index.html)
 
 - **Pigeon 2.0**
 
-    - API Documentation (v5: [Java](https://api.ctr-electronics.com/phoenix/release/java/com/ctre/phoenix/sensors/WPI_Pigeon2.html), [C++](https://api.ctr-electronics.com/phoenix/release/cpp/classctre_1_1phoenix_1_1sensors_1_1_w_p_i___pigeon2.html) | Pro: [Java](https://api.ctr-electronics.com/phoenixpro/release/java/com/ctre/phoenixpro/hardware/Pigeon2.html), [C++](https://api.ctr-electronics.com/phoenixpro/release/cpp/classctre_1_1phoenixpro_1_1hardware_1_1_pigeon2.html))
+    - API Documentation (v6: [Java](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/Pigeon2.html), [C++](https://api.ctr-electronics.com/phoenix6/release/cpp/namespacectre_1_1phoenix_1_1platform.html#ad0cac06e0817cbcfe20851a8619a19a8a7bd5c8d59e1daa07d31aa1d7116c5c64) | Pro: [Java](https://api.ctr-electronics.com/phoenixpro/release/java/com/ctre/phoenixpro/hardware/Pigeon2.html), [C++](https://api.ctr-electronics.com/phoenixpro/release/cpp/classctre_1_1phoenixpro_1_1hardware_1_1_pigeon2.html))
```
Comment (sciencewhiz) [inline source/docs/software/can-devices/third-party-devices.rst:50]: ditto

File: source/docs/software/can-devices/third-party-devices.rst (hunk 3 above)
Comment (Spaceman113138) [reply source/docs/software/can-devices/third-party-devices.rst]: Should be fixed now.

**Review by sciencewhiz — APPROVED** [review]

**Pull request merged**